.github/workflows/template-container-image-build.yml

```diff
@@ -0,0 +1,51 @@
+on:
+  workflow_call:
+      inputs:
+        image_to_push:
+          required: true
+          type: string
+        tag_name:
+          required: true
+          type: string
+        tag_latest:
+          required: false
+          type: boolean
+          default: false
+
+name: Container Image Publishing - Build
+
+jobs:
+  build:
+    name: Build Application
+    runs-on: ubuntu-latest
+    steps:
+      - name: Code Checkout
+        uses: actions/checkout@v2
+
+      - name: Log in to Registry
+        run: echo "${{ secrets.GITHUB_TOKEN }}" | docker login ghcr.io -u $ --password-stdin
+
+      - name: Build Docker Image
+        run: |
+          docker-compose build ${{ inputs.image_to_push }}
+
+      - name: Create Tag
+        run: |
+          GH_IMAGE_PATH="ghcr.io/${{ github.repository }}/${{ inputs.image_to_push }}"
+          GH_TAG=$GH_IMAGE_PATH:${{ inputs.tag_name }}
+        
+      - name: Tag Latest
+        if: ${{ inputs.tag_latest }}
+        run: |
+          GH_TAG=$GH_TAG $GH_IMAGE_PATH:latest
```

Comment:
This doesn't look quite right

If GH_TAG on line 35 was `ghcr.io/azure/azure-saas/asdk-admin:v1.0`, this would mean GH_TAG here on line 40 would become `ghcr.io/azure/azure-saas/asdk-admin:v1.0:latest` which isn't the correct syntax. After the ":", it should either be "latest" or "v1.0" (or whatever the input tag is. 

You'll need to issue two separate `docker tag` commands (line 48) to do this. with the two separate tag names.

Reply:
Not quite, the aim was to construct multiple tags inline so it would become `ghcr.io/azure/azure-saas/asdk-admin:v1.0 ghcr.io/azure/azure-saas/asdk-admin:latest`. This was a mistake, however, as that inline multiple tagging is only available for the build command now that I check.

This has been split into multiple independent operations and the image pushed with all tags.